defactor-com/sdk
# Feat/implement claim 165

## Implement `claim` function

### What does this PR do?

- Resolve #165

### Test Cases for `claim`

   - Failure
      - contract is paused
      - non-existed pool
      - the signer is the owner of the pool
      - Status is not `CLOSED` or `ACTIVE`
      - The pool is already claimed
      - The pool has no deposited rewards
      - The signer has no commit to the pool
   - Sucess
       - pool status is `CLOSED` 
       - pool status is `ACTIVE` 
       
### Steps to test

1. Configure the `test-util.ts` with the required contract address, pk's, etc
2. Run the test
3. Check the test results

### Test results

![imagen](https://github.com/defactor-com/sdk/assets/66583677/ab986935-725f-4f25-86d1-f7e13d5c0b79)
![image](https://github.com/defactor-com/sdk/assets/66583677/ef53605d-43d4-4477-b3ba-c00846cb591f")

#### CheckList
